qcodes/tests/dataset/test_dataset_in_memory.py

```diff
@@ -43,6 +46,53 @@ def test_dataset_in_memory_reload_from_db(
     compare_datasets(ds, loaded_ds)
 
 
+@settings(
+    deadline=None,
+    suppress_health_check=(HealthCheck.function_scoped_fixture,),
+    max_examples=10,
+)
+@given(
+    shape1=hst.integers(min_value=1, max_value=100),
+    shape2=hst.integers(min_value=1, max_value=100),
+)
+def test_dataset_in_memory_reload_from_db_2d(
+    meas_with_registered_param_2d, DMM, DAC, tmp_path, shape1, shape2
+):
+    meas_with_registered_param_2d.set_shapes(
+        {
+            DMM.v1.full_name: [shape1, shape2],
+        }
+    )
+    i = 0
+    with meas_with_registered_param_2d.run(
+        dataset_class=DataSetType.DataSetInMem
+    ) as datasaver:
+        for set_v in np.linspace(0, 25, shape1):
+            for set_v2 in np.linspace(0, 100, shape2):
+                DAC.ch1.set(set_v)
+                DAC.ch2.set(set_v2)
+                get_v = DMM.v1()
+                datasaver.add_result(
+                    (DAC.ch1, set_v), (DAC.ch2, set_v2), (DMM.v1, float(i))
+                )
+                i = i + 1
```

Comment:
unused, right? 
```suggestion
```

Reply:
No we are inserting a counter into the data rather then the output from the instrument to make sure that the data is written and read in the expected order